### ISSUE TITLE
How to best setup validation set

### ISSUE THREAD

**Issue by nikky4D** [layumi/Person_reID_baseline_pytorch#349]
My validation value is not going down. The loss is static mostly, and the accuracy on the validation set is 0. The training accuracy, however, is high. (see image below)

I thought this may be due to how I setup the splits. I setup my val such that my train and val have different IDs. My gallery also has a different set of ids. The queries all come from the ids in the gallery using images similar to train set. Is this the correct way to do this? I looked at prepare-*.py scripts and I can see that sometimes val is just a few samples from the train set, and has same ids in train set.  What do you recommend for setting up a val set?

![image](https://user-images.githubusercontent.com/7451106/216702276-322c620e-0762-4a01-be85-1c1be72af4af.png)

**Comment by taozeze** [comment]
我也遇到了这个问题 求郑大佬告知

**Comment by layumi** [comment]
Hi，@nikky4D @taozeze 

Since we apply a classification model during training, it would be good to directly leave out some samples per class for validation.  It is identical to the classification setting.

**Comment by nikky4D** [comment]
Thanks. But if I'm training with train_all flag, this combines the ids in val and train. Shouldn't I see better accuracy then for the validation?
